testing/a11y/src/Component.cy.tsx

```diff
@@ -6,6 +6,34 @@ import * as React from 'react';
 import * as allExamples from 'examples';
 import { ThemeProvider } from '@itwin/itwinui-react';
 
+// Axe config rules docs: https://www.deque.com/axe/core-documentation/api-documentation/#api-name-axeconfigure
+// Axe rules that apply to multiple examples
+type commonRulesKeys = 'ignore_color_contrast_for_select_placeholder';
+const commonRules: Record<commonRulesKeys, Record<string, any>> = {
+  ignore_color_contrast_for_select_placeholder: {
+    id: 'color-contrast',
+    enabled: true,
+    selector: ':not(._iui3-placeholder > span._iui3-content)',
+  },
```

Comment:
oh this is too much code and too hacky. now i'm thinking we should just fix the contrast.

my only concern is there will not be enough distinction between placeholder text and regular text.

cc @FlyersPh9 for opinions on visuals

Reply:
> oh this is too much code and too hacky. now i'm thinking we should just fix the contrast.

This PR was WIP. The PR was going to be ready for review in a few minutes, during which I was going to leave a PR comment saying I am open to suggestions to improve this code, if needed. But you were quicker than me! 😅

> my only concern is there will not be enough distinction between placeholder text and regular text.

I intentionally added very selective `selector` to only ignore color contrast only for the placeholder text and not any other text.

Reply:
> > my only concern is there will not be enough distinction between placeholder text and regular text.
> 
> I intentionally added a very selective `selector` to only ignore `color-contrast` only for the placeholder text and not any other text.

no i'm saying if we increase contrast for the placeholder, then it will look very similar to the regular text (for the end user)

Reply:
> oh this is too much code and too hacky. now i'm thinking we should just fix the contrast.

This is the code to ignore the color contrast so that it can pass our GitHub checks?  It might be a lot and hacky, but I'd value the end result more over the test code being clean.

> my only concern is there will not be enough distinction between placeholder text and regular text.

Definitely a concern.  I tried changing the color from `--iui-color-text-disabled` to `--iui-color-text-muted` which does pass contrast scores with a 7 in light theme & 4.62 in dark theme, but is _very_ easy to mistake for regular text.

**Current implementation:**
![Screenshot 2023-08-10 at 2 53 54 PM](https://github.com/iTwin/iTwinUI/assets/849817/a992af96-3ea9-4adc-9505-aa9df6adfae3)

**Using muted text color:**
![Screenshot 2023-08-10 at 2 52 17 PM](https://github.com/iTwin/iTwinUI/assets/849817/fd586d0c-ffd3-431f-b0ad-47ab9fa42059)

**Actual text:**
![Screenshot 2023-08-10 at 2 52 33 PM](https://github.com/iTwin/iTwinUI/assets/849817/e9867183-cf56-4b42-8f81-fe6f28c58fc7)

We could consider adding a new text color variable specifically for placeholder text, but the dark disabled text color is already really close to the 4.5 minimum requirement.

Reply:
for the placeholder, could we try a color that is the bare minimum (4.5)? and if that's still to close to regular text, can we try bumping the contrast of regular text?

Reply:
> This is the code to ignore the color contrast so that it can pass our GitHub checks?

Yes, this code is to ignore the reported `color-contrast` violation for just the placeholder texts of only the examples for `Select`

> We could consider adding a new text color variable specifically for placeholder text, but the dark disabled text color is already really close to the 4.5 minimum requirement.

Just including the exact contrast ratio for our reference. The contrast ratio for the placeholder text is `2.93`.

![image](https://github.com/iTwin/iTwinUI/assets/45748283/a27124a2-f235-4ee9-a400-ece65bb2749b)

Reply:
How about this?  I didn't try high contrast yet.

`light.scss`
```diff
- --iui-color-text-disabled: #{hex($fg-700)};
+ --iui-color-text-disabled: #{hex($fg-600)}; 
```

`dark.scss`
```diff
- --iui-color-text-muted: #{hex($fg-500)};
+ --iui-color-text-muted: #{hex($fg-400)};
- --iui-color-text-disabled: #{hex($fg-700)};
+ --iui-color-text-disabled: #{hex($fg-500)};
```

**Before:**
![Screenshot 2023-08-10 at 3 39 08 PM](https://github.com/iTwin/iTwinUI/assets/849817/04da137d-55fd-4ae9-a185-8b164ac1d625) ![Screenshot 2023-08-10 at 3 28 22 PM](https://github.com/iTwin/iTwinUI/assets/849817/d234920f-9ae3-49d2-94b2-6c82b11fe23e)

<table>
<tr>
 <td>Variable
 <td>Light
 <td>Dark
<tr>
 <td>Text
 <td>15.52
 <td>8.13
<tr>
 <td>Muted
 <td>7
 <td>4.62
<tr>
 <td>Disabled
 <td><strong>4.11</strong>
 <td><strong>2.93</strong>
</table>

**After:**
![Screenshot 2023-08-10 at 3 40 34 PM](https://github.com/iTwin/iTwinUI/assets/849817/5eba73c7-4534-4b5c-ac90-be9ca9c0b4fb) ![Screenshot 2023-08-10 at 3 27 28 PM](https://github.com/iTwin/iTwinUI/assets/849817/e9b53dae-8758-43f5-9acc-f22dd5228e38)

<table>
<tr>
 <td>Variable
 <td>Light
 <td>Dark
<tr>
 <td>Text
 <td>15.52
 <td>8.13
<tr>
 <td>Muted
 <td>7
 <td>5.68
<tr>
 <td>Disabled
 <td>5.32
 <td>4.62
</table>

Reply:
> ```diff
> - --iui-color-text-muted: #{hex($fg-500)};
> + --iui-color-text-muted: #{hex($fg-400)};
> ```

isn't that bringing it even closer to regular text?

also would need to see in the context of a select. hard to tell from just looking at colors.

> didn't try high contrast yet.

just remember it's ok if placeholder doesn't pass 7:1 because it still meets minimum requirements (which is all that's needed from a legal perspective)

> 4.11 ⚠️ | 2.93 ⚠️

also remember disabled elements are exempt from contrast requirements

Reply:
> How about this?

@FlyersPh9 Thanks for the detailed info! It's nice that the contrast ratio minimum of 4.5 is met. However, I find very little distinction between (normal & muted), and even lesser between (muted & disabled) with the new colors.

> just remember it's ok if placeholder doesn't pass 7:1 because it still meets minimum requirements (which is all that's needed from a legal perspective)

@mayank99, to confirm, is the legal minimum `4.5` for AA? If so, are we are not clearing the minimum for AA? Because the contrast ratios are `4.11` and `2.93`?

Reply:
yes https://webaim.org/articles/contrast/#sc143

Reply:
> @mayank99, to confirm, is the legal minimum `4.5` for AA? 

> yes https://webaim.org/articles/contrast/#sc143

Thanks for the link.

>  If so, are we are not clearing the minimum for AA? Because the contrast ratios are `4.11` and `2.93`?

Ohh, I meant to say that you had said that we *are* meeting the minimum requirements (I assume AA (4.5:1) is the legal minimum) for the placeholder. So, it is okay if we don't meet the more ambitious AAA requirement (7:1). 

> just remember it's ok if placeholder doesn't pass 7:1 because it still meets minimum requirements (which is all that's needed from a legal perspective)

But since the old contrast ratios for disabled are `4.11` and `2.93`, the placeholder is *not* meeting the minimum AA (4.5:1) legal requirement.

Or maybe I am misunderstanding something?

Reply:
> > ```diff
> > - --iui-color-text-muted: #{hex($fg-500)};
> > + --iui-color-text-muted: #{hex($fg-400)};
> > ```
> isn't that bringing it even closer to regular text?

Yes, this makes muted closer to regular.  But muted had to change because disabled is now equal to what muted was.

> i am also confused about which color we're using (disabled vs muted) for placeholder text.

Placeholder text is currently using `--iui-color-text-disabled`.

> > to confirm, is the legal minimum `4.5` for AA? If so, are we are not clearing the minimum for AA? Because the contrast ratios are `4.11` and `2.93`?
> yes https://webaim.org/articles/contrast/#sc143

I swear that we looked this up and placeholder text did not need to meet 4.5, but as I'm googling this now [it appears that placeholder text DOES need to meet 4.5](https://www.w3.org/WAI/WCAG21/Understanding/contrast-minimum.html#:~:text=including%20placeholder%20text).

Reply:
placeholder needs to meet 4.5, disabled/inactive doesn't.

the 4.5 ratio is the only requirement because level AA is what is used in the laws worldwide. AAA is nice to have, but is not a legal requirement anywhere in the world afaict

Reply:
In our case, since we are using the disabled colors for the placeholder, looks like we still have to meet this 4.5 contrast ratio requirement. Because the placeholder/Select is still *user interactable*. ([link](https://www.w3.org/WAI/WCAG21/Understanding/contrast-minimum.html))

![image](https://github.com/iTwin/iTwinUI/assets/45748283/3f14496d-2888-4665-a0ee-8380fc5c3c81)

So we may need to create that new variable, or modify the existing colors to meet that requirement.

Reply:
yeah, that's why i've been trying to suggest using a completely new color specifically for placeholder.

Reply:
Take a look at #1485.